### DISCUSSION TITLE
Vec3 world_to_viewport

### DISCUSSION THREAD

**Discussion by DasLixou** [bevyengine/bevy#14583] [Q&A] [answered]
Given a world coordinate and putting it through world_to_viewport, I only get a Vec2 screen coordinate. Is there a method to get the depth, too?
Seems like world_to_ndc is Vec3, but is the Z value between 0 and 1 corresponding to near and far plane?

**Comment by bugsweeper** [comment]
Yes, world_to_viewport in it's body uses world_to_ndc, as you can see in [comment](https://github.com/bevyengine/bevy/blob/fc2f564c6f205163b8ba0f3a0bd40fe20739fa7a/crates/bevy_render/src/camera/camera.rs#L351C5-L352C68)
and in it's body there is check for [that](https://github.com/bevyengine/bevy/blob/fc2f564c6f205163b8ba0f3a0bd40fe20739fa7a/crates/bevy_render/src/camera/camera.rs#L364C8-L367C10)

**Reply by DasLixou** [reply, marked as answer]
sorry, didnt expect an answer anymore 😅 - I already added #14590 which gives me the required maths
